ephios/event_management/mail.py

```diff
@@ -37,19 +37,35 @@ def new_event(event):
     )
 
     for user in users:
-        message = EmailMultiAlternatives(
-            to=[user.email], subject=subject, body=text_content, reply_to=responsible_persons_mails
-        )
-        message.attach_alternative(html_content, "text/html")
-        messages.append(message)
+        if user.preferences["notifications__new_event"]:
+            message = EmailMultiAlternatives(
+                to=[user.email],
+                subject=subject,
+                body=text_content,
+                reply_to=responsible_persons_mails,
+            )
+            message.attach_alternative(html_content, "text/html")
+            messages.append(message)
     mail.get_connection().send_messages(messages)
 
 
 def participation_state_changed(participation: AbstractParticipation):
     messages = []
 
     # send mail to the participant whose participation has been changed
-    if participation.participant.email is not None and participation.state in (
+    mail_requested = participation.participant.email is not None
+    if participation.get_real_instance_class() == LocalParticipation:
+        local_participation = participation.get_real_instance()
```

Comment:
my understanding was that participation already is of the correct type when using the polymorphic managers. ([Doc](https://django-polymorphic.readthedocs.io/en/stable/api/polymorphic.models.html?highlight=get_real_instance#polymorphic.models.PolymorphicModel.get_real_instance))

Is that not given here?

Reply:
just checked, this is given here